### PR TITLE
liqui: fetchOrders: fixed a bug when order erroneously marked as closed

### PR DESCRIPTION
**Pull request by mkutny**
Might be related to #1801 as the code is actually the same but as well it might happen that #1801 shows the effect of another bug.

Steps to reproduce:
```
> liqui.fetchOrders()
{ }
> liqui.createLimitBuyOrder('BTC/USDT', 1, 1)
> liqui.fetchOrders()
> liqui.orders
{...status: 'open' }
> liqui.fetchOrders('ETH/BTC')
> liqui.orders
{...status: 'closed' }
> liqui.fetchOrders('BTC/USDT')
> liqui.orders
{...status: 'open' }
```

I moved cache handling to a separate function so it's easier to share between exchanges.

I also removed the following check:
https://github.com/ccxt/ccxt/blob/07978ac2289a755e66d02da17f11ba8fe0d3ebd1/js/liqui.js#L533-L534

It looks to me that it duplicates what we do few lines above:
https://github.com/ccxt/ccxt/blob/07978ac2289a755e66d02da17f11ba8fe0d3ebd1/js/liqui.js#L526

Can't guess a case when it would be needed but letting you know if it was done on a purpose.

### CODE REVIEW

File: js/liqui.js [hunk 1]
```diff
@@ -508,35 +508,31 @@ module.exports = class liqui extends Exchange {
...
                 let order = this.orders[id];
+                if (typeof symbol !== 'undefined' && symbol !== order['symbol']) {
+                    // fetched a particular symbol but it doesn't match the order in the cache -> won't update
+                    continue;
+                }
+                // order is cached but not opened -> mark as closed
                 if (order['status'] === 'open') {
                     order = this.extend (order, {
-                        'status': 'closed',
+                        'status': 'closed', // likewise it might have been canceled
```
Comment (kroitor) [inline js/liqui.js]: Yep, but only if the order wasn't closed by this instance of Liqui. I mean, when we cancelOrder, we cache the canceled status as well, so upon fetchOrders we don't enter this codepath at all. So, there's no way for the order status to change, unless it's changed externally. The cache has to be properly saved and restored between the runs to sustain it, of course.

File: js/liqui.js [hunk 2]
```diff
@@ -508,35 +508,31 @@ module.exports = class liqui extends Exchange {
...
+            // - order was closed or cancelled -> update cache
+            // - symbol mismatch (e.g. cached BTC/USDT, fetched ETH/USDT) -> skip
             let id = cachedOrderIds[k];
-            if (id in openOrdersIndexedById) {
-                this.orders[id] = this.extend (this.orders[id], openOrdersIndexedById[id]);
-            } else {
+            if (!(id in openOrdersIndexedById)) {
+                // cached order is not in open orders array
                 let order = this.orders[id];
+                if (typeof symbol !== 'undefined' && symbol !== order['symbol']) {
```
Comment (kroitor) [inline js/liqui.js]: I'm totally ok with merging this pull request, and wanted to thank you again for your participation in this project.

This is not an issue nor a problem at all, just wanted to share a couple of thoughts. When I'm looking at this line, here's what I'm thinking of: we are lucky that it does the job for Liqui, as their order ids (id-spaces) rarely intersect. However, I can foresee problems with this type of design if we run into an exchange that has symbol-specific ids, like you said earlier before... And from here, to make this method really generic we might need two types of cache:
- an *orders indexed by id* cache (current type)
- an *orders grouped by symbol indexed by id* cache (missing now)

Again these thought are not related to the pull request, I'm merging it very soon. But apparently this method is going to migrate to the base class someday, so, we might take a few considerations along the way )

File: js/liqui.js (hunk 1 above)
Comment (mkutny) [reply js/liqui.js]: You're right. I indeed used different sessions.

File: js/liqui.js (hunk 2 above)
Comment (mkutny) [reply js/liqui.js]: Yeah, it's a good catch, as you would say )

By their nature orders belong to markets even if they share exchange-wide id space. And if we look at it from this perspective I think we won't need 'orders indexed by id' at all, 'orders grouped by symbol indexed by id' are generic enough to serve all cases, right?

**Pull request merged**